task-driver/integration/helpers.rs

```diff
@@ -94,21 +99,38 @@ pub async fn new_wallet_in_darkpool(client: &ArbitrumClient) -> Result<(Wallet,
     let blinder_seed = Scalar::random(&mut rng);
     let share_seed = Scalar::random(&mut rng);
 
-    let wallet = empty_wallet_from_seed(blinder_seed, share_seed);
-    allocate_wallet_in_darkpool(&wallet, client).await?;
+    let mut wallet = empty_wallet_from_seed(blinder_seed, share_seed);
+    allocate_wallet_in_darkpool(&mut wallet, client).await?;
 
     Ok((wallet, blinder_seed, share_seed))
 }
 
-/// Create a wallet in the contract state
-pub async fn allocate_wallet_in_darkpool(wallet: &Wallet, client: &ArbitrumClient) -> Result<()> {
+/// Create a wallet in the contract state and update the Merkle path on the
+/// wallet
+pub async fn allocate_wallet_in_darkpool(
+    wallet: &mut Wallet,
+    client: &ArbitrumClient,
+) -> Result<()> {
     let share_comm = wallet.get_private_share_commitment();
 
     let mut proof = dummy_valid_wallet_create_bundle();
     proof.statement.public_wallet_shares = wallet.blinded_public_shares.clone();
     proof.statement.private_shares_commitment = share_comm;
 
-    client.new_wallet(proof).await.map_err(Into::into)
+    client.new_wallet(proof).await.map_err(Into::<Report>::into)?;
```

Comment:
nit: I think you can just do `client.new_wallet(proof).await?` and any type implementing `Error` will automatically be turned `Into` a `Report`

but if you tried this already and it didn't work then nvm, must be misremembering